lib/core/CDataFrame.cc

```diff
@@ -410,7 +408,7 @@ CDataFrame::TRowFuncVecBoolPr
 CDataFrame::parallelApplyToAllRows(std::size_t numberThreads,
                                    std::size_t beginRows,
                                    std::size_t endRows,
-                                   TRowFunc func,
+                                   TRowFunc& func,
```

Comment:
It seems dodgy to pass this by lvalue reference here, as when it's passed on again it's passed on with `std::move(func)`.  Should it be an rvalue reference here too?

Same for `sequentialApplyToAllRows`.

Reply:
This can be an rvalue reference, but it not in `sequentialApplyToAllRows` because I don't move from there and pass down by reference to another function in a loop.

(It ought to be equivalent passing by lvalue reference because the upshot is it'll still get moved into `SCallableWithBoundState`, but perhaps passing by rvalue reference makes the intention clearer.)